compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/EncodeV1.hs

```diff
@@ -29,15 +31,23 @@ import qualified Proto3.Suite as P (Enumerated (..))
 -- otherwise always be wrapped in `Just` at their call sites.
 type Just a = Maybe a
 
+-- package-global state that encodePackageRef requires
+type PackageRefCtx = S.Set PackageId
```

Comment:
I'd prefer a `Map PackageId Int` instead of using `lookupIndex`.

Reply:
In that case we lose the easy preservation of the important invariant that the values are always `[0..n]` for some n; what about making the context passed around `PackageId -> Word64` and have only the set-maker care about `lookupIndex`?

Reply:
⬆️ as in a1c30a08da389207e0db44cf80b63796cbce425e.